src/Microsoft.Framework.Runtime/Servicing/Breadcrumbs.cs

```diff
@@ -0,0 +1,117 @@
+            {
+                Trace.TraceInformation("Breadcrumbs for servicing will not be written because the breadcrumbs folder ({0}) does not exist.", breadcrumbsFolder);
+            }
+        }
+
+        public static bool IsPackageServiceable(IPackage package)
+        {
+            // TODO: Figure out what makes a package serviceable
+            return false;
+        }
+
+        public void CreateBreadcrumb(string packageId, SemanticVersion packageVersion)
+        {
+            if (string.IsNullOrWhiteSpace(_breadcrumbsFolder))
+            {
+                return;
+            }
+
+            // Create both files for now until we get clear instructions about the format of the name
+            CreateBreadcrumbfile(packageId);
+            CreateBreadcrumbfile(packageId + "." + packageVersion);
+        }
+
+        public void CreateRuntimeBreadcrumb()
+        {
+#if ASPNET50
+            var runtimeAssembly = typeof(Breadcrumbs).Assembly;
+#else
+            var runtimeAssembly = typeof(Breadcrumbs).GetTypeInfo().Assembly;
+#endif
+
+            var version = runtimeAssembly
+                ?.GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                ?.InformationalVersion;
+
+            if (!string.IsNullOrWhiteSpace(version))
+            {
+                var semanticVersion = new SemanticVersion(version);
+
+                CreateBreadcrumb(runtimeAssembly.GetName().Name, semanticVersion);
+            }
+        }
+
+        private static string ResolveBreadcrumbsFolder()
+        {
+            var programDataFolder = Environment.GetEnvironmentVariable("ProgramData");
+            if (string.IsNullOrWhiteSpace(programDataFolder))
+            {
+                return null;
+            }
+
+            string breadcrumbsFolder = Path.Combine(
+                programDataFolder,
+                "Microsoft",
+                "KRE",
+                "BreadcrumbStore");
+
+            return breadcrumbsFolder;
+        }
+
+        private void CreateBreadcrumbfile(string fileName)
+        {
+            string fullFilePath = Path.Combine(_breadcrumbsFolder, fileName);
+
+            try
+            {
+                if (!File.Exists(fullFilePath))
+                {
+                    File.Create(fullFilePath).Dispose();
+                    Trace.TraceInformation("Wrote servicing breadcrumb for {0}", fileName);
+                }
+            }
+            catch (UnauthorizedAccessException exception)
+            {
+                LogBreadcrumbsCreationFailure(fileName, exception);
+            }
+            catch (DirectoryNotFoundException exception)
+            {
+                LogBreadcrumbsCreationFailure(fileName, exception);
+            }
+        }
+
+        private static void LogBreadcrumbsCreationFailure(string fileName, Exception exception)
+        {
+            Trace.TraceInformation("Failed to write servicing breadcrumb for {0} because of {1}", fileName, exception);
```

Comment:
Change to `... because an exception was thrown: {1}`. Otherwise there's no way the logged text would make sense.